### ISSUE TITLE
Check pubsub subscription and resub if needed

### ISSUE THREAD

**Issue by v-stickykeys** [ceramicnetwork/js-ceramic#1488]
Somehow the IPFS node used by gateway.ceramic.network got unsubscribed from pubsub today.

This happened without an error being thrown, but we were able to tell because we got an alert from Grafana saying that the gateway was no longer receiving pubsub messages.

Then I confirmed it was unsubscribed

```sh
$ curl -X POST https://ipfs-ceramic-public-mainnet-internal.ceramic.network:5011/api/v0/pubsub/peers?arg=/ceramic/mainnet
{"Strings":["QmXALVsXZwPWTUbsT8G6VVzzgTJaAWRUD7FWL5f7d5ubAL","QmUiF8Au7wjhAF9BYYMNQRW5KhY7o8fq4RUozzkWvHXQrZ","QmRNw9ZimjSwujzS3euqSYxDW9EHDU5LB3NbLQ5vJ13hwJ"]}

$ curl -X POST https://ipfs-ceramic-public-mainnet-internal.ceramic.network:5011/api/v0/pubsub/ls
{"Strings":[]}
```

I propose we add back the logic to periodically resubscribe and at the same time investigate why the unsub happened in the first place.

**Comment by v-stickykeys** [comment]
Also cc @zachferland bc I think you worked on the last iteration of this

Not clear to me what happened in the case described above.

It could be:
- IPFS node unsubscribed without being told to by the client. Though I have not found evidence to suggest this is possible/would happen
- IPFS node response sent an `end` event and the client unsubscribed based on logic here https://github.com/ipfs/js-ipfs/blob/b5470d40ea455069f3f3bd7ab3fb42d7c08926b4/packages/ipfs-http-client/src/pubsub/subscribe.js#L67 . I don't know why the stream would end though.

Fyi neither nodes restarted during this time

**Comment by zachferland** [comment]
Think we talked about this a bit last time, but the ideal solution would be to listen for onEnd and onError, or have ipfs client have an unsubscribe event. Assuming onEnd would not happen definitely wont be a safe assumption, probably can happen for many valid reasons. But I think we wanted the periodic resub removed at least at first to better see the behavior and see if the the fix was working. At the moment, adding that back is the best path. 

But thats great that the client is cleaning up subscriptions onEnd (expected behavior), wasnt the case before, at least we have some visibility now.

**Comment by zachferland** [comment]
I can add the story for further improvements

**Comment by v-stickykeys** [comment]
> Assuming onEnd would not happen definitely wont be a safe assumption, probably can happen for many valid reasons.

Makes sense @zachferland . However the problem is I don't yet know why it did end 😆 

But yeah would be great if you can create an issue or submit pr to ipfs team so we can pass the onEnd

**Comment by v-stickykeys** [comment]
@oed This doesn't feel urgent (in the sense that we've only seen it be a problem once) but should probably be done soon if possible. Added you but anyone could do this

**Comment by oed** [comment]
Thanks @v-stickykeys, I assigned you @zachferland since I think you have the best idea of what's needed here. Feel free to just spec this and maybe reassign to @smrz2001
